### PR TITLE
[feature](executor) Use the  BE info to set pipeline variables. 

### PR DESCRIPTION
**Pull request by Mryange**
# Proposed changes

In [ pr19121 ](https://github.com/apache/doris/pull/19121) , using heartbeat with a five-second interval. 
Now, using diskInfo.With a one-minute interval.
Used when enablePipelineEngine = true to represent parallelExecInstanceNum.Specifically, if enablePipelineEngine = true and pipelineParallelExecInstanceNum = 0, then half of the current CPU cores will be used.
```
mysql [(none)]>set pipeline_parallel_fragment_exec_instance_num = 0;
Query OK, 0 rows affected (0.01 sec)
```

## Problem summary
First test record （without fuzzy ,ParallelExecInstanceNum  = 16 ）
image
image
image
No modifications, retesting.
image
same as https://github.com/apache/doris/pull/19212

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

File: gensrc/thrift/MasterService.thrift [hunk 1]
```diff
@@ -81,6 +81,7 @@ struct TDisk {
     6: optional i64 path_hash
     7: optional Types.TStorageMedium storage_medium
     8: optional Types.TSize remote_used_capacity
+    9: optional i32 num_cores
```
Comment (HappenLee) [inline gensrc/thrift/MasterService.thrift:84]: the message should not set in TDisk

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 2]
```diff
@@ -479,7 +480,7 @@ public class SessionVariable implements Serializable, Writable {
      * 1 means disable this feature
      */
     @VariableMgr.VarAttr(name = PARALLEL_FRAGMENT_EXEC_INSTANCE_NUM, fuzzy = true)
-    public int parallelExecInstanceNum = 1;
+    public int parallelExecInstanceNum = 0;
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:483]: we need a new session variable to set pipeline parallel

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
In some cases, the "order by"  contains duplicates, so a portion of the regression test was modified.

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
There are two related PRs here, the first one sets using the information collected by BE, and the second one tests with fuzzy and pipeline enabled (because this PR is even-numbered).
https://github.com/apache/doris/pull/19324
https://github.com/apache/doris/pull/19345

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Pull request closed without merging**